### PR TITLE
add MOI.get for BarrierIterations

### PR DESCRIPTION
**Pull request by lkapelevich**

### CODE REVIEW

File: src/MOI_wrapper.jl [hunk 1]
```diff
@@ -28,6 +29,7 @@ function Solution()
         NaN,
         NaN,
         NaN,
+        0,
```
Comment (blegat) [inline src/MOI_wrapper.jl:32]: This should probably be reset in `empty!`

Comment (lkapelevich) [reply src/MOI_wrapper.jl:32]: I think this https://github.com/jump-dev/ECOS.jl/blob/master/src/MOI_wrapper.jl#L135 takes care of that

**Review by blegat — APPROVED** [review]

**Comment by chriscoey** [comment]
@blegat could you merge? I don't have permissions

**Comment by lkapelevich** [comment]
would someone mind creating a new tag?

**Comment by odow** [comment]
https://github.com/JuliaRegistries/General/pull/41493

**Pull request merged**